sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1665,7 +1665,7 @@ class Dataset[T] private[sql](
   }
 
   /**
-   * Returns a new [[Dataset]] by adding a column with metadata.
+   * Returns a new [[DataFrame]] by adding a column with metadata.
```

Comment:
these were technically correct; a `DataFrame` is a `Dataset[Row]`. I would revert these changes